tensorflow_addons/losses/sparsemax_loss_test.py

```diff
@@ -204,42 +198,48 @@ def test_sparsemax_loss_of_inf(self, dtype=None):
         _, tf_loss_mix = _tf_sparsemax_loss(z_mix, q, dtype)
         self.assertAllEqual([np.nan, np.nan, np.nan, np.nan], tf_loss_mix)
 
-    def test_constant_add(self, dtype=None):
-        """check sparsemax-loss proposition 3."""
-        random = np.random.RandomState(4)
 
-        z = random.uniform(low=-3, high=3, size=(test_obs, 10))
-        c = random.uniform(low=-3, high=3, size=(test_obs, 1))
-        q = np.zeros((test_obs, 10))
-        q[np.arange(0, test_obs), np.random.randint(0, 10, size=test_obs)] = 1
+@pytest.mark.parametrize("dtype", ["float32", "float64"])
+def test_constant_add(dtype):
+    """check sparsemax-loss proposition 3."""
+    random = np.random.RandomState(4)
 
-        _, tf_loss_zpc = _tf_sparsemax_loss(z + c, q, dtype)
-        _, tf_loss_z = _tf_sparsemax_loss(z, q, dtype)
+    z = random.uniform(low=-3, high=3, size=(test_obs, 10))
+    c = random.uniform(low=-3, high=3, size=(test_obs, 1))
+    q = np.zeros((test_obs, 10))
+    q[np.arange(0, test_obs), np.random.randint(0, 10, size=test_obs)] = 1
 
-        self.assertAllCloseAccordingToType(
-            tf_loss_zpc, tf_loss_z, float_atol=5e-6, float_rtol=5e-6
-        )
+    _, tf_loss_zpc = _tf_sparsemax_loss(z + c, q, dtype)
+    _, tf_loss_z = _tf_sparsemax_loss(z, q, dtype)
 
-    def test_sparsemax_loss_zero(self, dtype=None):
-        """check sparsemax-loss proposition 5."""
-        random = np.random.RandomState(6)
+    test_utils.assert_allclose_according_to_type(
+        tf_loss_zpc, tf_loss_z, float_atol=5e-6, float_rtol=5e-6
+    )
 
-        # construct z and q, such that z_k >= 1 + max_{j!=k} z_k holds for
-        # delta_0 = 1.
-        z = random.uniform(low=-3, high=3, size=(test_obs, 10))
-        z[:, 0] = np.max(z, axis=1) + 1.05
 
-        q = np.zeros((test_obs, 10))
-        q[:, 0] = 1
+@pytest.mark.parametrize("dtype", ["float32", "float64"])
+def test_sparsemax_loss_zero(dtype):
+    """check sparsemax-loss proposition 5."""
+    random = np.random.RandomState(6)
 
-        tf_loss_op, tf_loss_out = _tf_sparsemax_loss(z, q, dtype)
-        tf_sparsemax_op, tf_sparsemax_out = self._tf_sparsemax(z, dtype)
+    # construct z and q, such that z_k >= 1 + max_{j!=k} z_k holds for
+    # delta_0 = 1.
+    z = random.uniform(low=-3, high=3, size=(test_obs, 10))
+    z[:, 0] = np.max(z, axis=1) + 1.05
+
+    q = np.zeros((test_obs, 10))
+    q[:, 0] = 1
+
+    tf_loss_op, tf_loss_out = _tf_sparsemax_loss(z, q, dtype)
 
-        self.assertAllCloseAccordingToType(np.zeros(test_obs), tf_loss_out)
-        self.assertShapeEqual(np.zeros(test_obs), tf_loss_op)
+    tf_sparsemax_op = sparsemax(z.astype(dtype))
+    tf_sparsemax_out = tf_sparsemax_op
```

Comment:
It's not needed to have two variable to represent the same thing. It made sense in graph mode where we had ops and tensors, but in eager mode it's not needed.

Reply:
Got it, thanks @gabrieldemarmiesse